pyccl/core.py

```diff
@@ -913,7 +913,8 @@ def compute_nonlin_power(self):
         if (mps != 'emu') and (mps is not None):
             self.compute_linear_power()
 
-        if mps == "camb":
+        if mps == "camb" and self._has_pk_nl:
+            # Already computed
```

Comment:
I think this is gonna make the tests fail.

Reply:
Fixed.